File: 프로그래머스/lv3/42628. 이중우선순위큐/이중우선순위큐.js
function solution(operations) {
    let result = []
    operations.forEach(op => {
        const [command, data] = op.split(' ')
        if(command === 'I') {
            result.push(parseInt(data))
        } else if(command === 'D'){
            result = result.sort((a,b) => a-b)
            if(result.length){
                if(data === '1') result.pop()
                else result.shift()
            }
        }
    })
    if(!result.length) return [0, 0]
    else {
        const sorted = result.sort((a,b) => a-b)
        return [sorted.pop(), sorted.shift()]
    }
}

File: 프로그래머스/lv3/42628. 이중우선순위큐/이중우선순위큐.py
import heapq

def solution(operations):
    heap = []
    for op in operations:
        command, data = op.split(' ')
        data = int(data)
        if command == 'I':
            heapq.heappush(heap, data)
        elif len(heap) == 0:
            continue
        elif data == -1:
            heapq.heappop(heap)
        else:
            heap = [-el for el in heap]
            heapq.heapify(heap)
            heapq.heappop(heap)
            heap = [-el for el in heap]
            heapq.heapify(heap)
    
    heap.sort()
    return [0, 0] if len(heap) == 0 else [heap[len(heap) -1], heap[0]]
        